Tests/SonarScanner.MSBuild.Shim.Test/PathHelperTests.cs

```diff
@@ -29,164 +30,162 @@ namespace SonarScanner.MSBuild.Shim.Test
             }).Should().BeNull();
-        }
 
         [TestMethod]
-        public void GetCommonRoot_WhenCommonPath_ReturnsTheLongestCommonPart()
-        {
-            PathHelper.GetCommonRoot(new[]
+        public void BestCommonPrefix_WhenNoCommonPath_SameCountInEachGroup_ReturnsNull() =>
+            PathHelper.BestCommonPrefix(new[]
             {
-                new DirectoryInfo("C:\\Foo"),
-                new DirectoryInfo("C:\\Foo\\Bar"),
-                new DirectoryInfo("C:\\Foo\\FooBar"),
-            }).FullName.Should().Be("C:\\Foo");
-        }
+                new DirectoryInfo(@"C:\"),
+                new DirectoryInfo(@"C:\Dir"),
+                new DirectoryInfo(@"D:\DirA"),
+                new DirectoryInfo(@"D:\DirB\SubDir"),
+                new DirectoryInfo(@"Z:\"),
+                new DirectoryInfo(@"Z:\Dir"),
+            }).Should().BeNull();
 
         [TestMethod]
-        public void GetCommonRoot_WhenCommonPathOfFiles_ReturnsTheLongestCommonPart()
-        {
-            PathHelper.GetCommonRoot(new[]
+        public void BestCommonPrefix_WhenNoCommonPath_SameCountInMostCommonGroup_ReturnsNull() =>
+            PathHelper.BestCommonPrefix(new[]
             {
-                new DirectoryInfo("C:\\Foo.cs"),
-                new DirectoryInfo("C:\\Foo\\Bar.cs"),
-                new DirectoryInfo("C:\\Foo\\FooBar.cs"),
-            }).FullName.Should().Be("C:\\");
-        }
+                new DirectoryInfo(@"C:\Temp"),
+                new DirectoryInfo(@"D:\ThreeTimes\A"),
+                new DirectoryInfo(@"D:\ThreeTimes\B"),
+                new DirectoryInfo(@"D:\ThreeTimes\C"),
+                new DirectoryInfo(@"E:\AlsoThreeTimes\A"),
+                new DirectoryInfo(@"E:\AlsoThreeTimes\B"),
+                new DirectoryInfo(@"E:\AlsoThreeTimes\C"),
+            }).Should().BeNull();
 
         [TestMethod]
-        public void GetParts_WhenNull_ThrowsArgumentNullException()
-        {
-            // Arrange
-            // Act
-            Action act = () => PathHelper.GetParts(null);
+        public void BestCommonPrefix_WhenNoCommonPath_ReturnsMostCommonOne_Simple() =>
+            PathHelper.BestCommonPrefix(new[]
+            {
+                new DirectoryInfo(@"C:\Temp"),
+                new DirectoryInfo(@"D:\WorkDir\Project"),
+                new DirectoryInfo(@"D:\WorkDir\Project.Tests"),
+            }).FullName.Should().Be(@"D:\WorkDir");
 
-            // Assert
-            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("directoryInfo");
-        }
+        [TestMethod]
+        public void BestCommonPrefix_WhenNoCommonPath_ReturnsMostCommonOne_Complex() =>
+            PathHelper.BestCommonPrefix(new[]
+            {
+                new DirectoryInfo(@"C:\Temp"),
+                new DirectoryInfo(@"D:\ThreeTimes\A"),
+                new DirectoryInfo(@"D:\ThreeTimes\B"),
+                new DirectoryInfo(@"D:\ThreeTimes\C"),
```

Comment:
if it's complex, let's improve
```suggestion
                new DirectoryInfo(@"D:\ThreeTimes\A1\A2\A3"),
                new DirectoryInfo(@"D:\ThreeTimes\A1\B"),
                new DirectoryInfo(@"D:\ThreeTimes\A"),
```

Reply:
The `ReturnsTheLongestCommonPart` aspect is tested in the original `BestCommonPrefix_WhenCommonPath_ReturnsTheLongestCommonPart`. This one does `_WhenNoCommonPath_ReturnsMostCommonOne` so I'd keep it focused on the problem it asserts.

It's `Complex` compared it's `Simple` counterpart where there was race between two roots. Here, 3 are present.